lib/devise_two_factor/spec_helpers/two_factor_authenticatable_shared_examples.rb

```diff
@@ -44,11 +44,25 @@
       expect(subject.valid_otp?(otp)).to be true
     end
 
+    it 'does not validate a previously precisely correct OTP' do
```

Comment:
I would probably move these new specs into a separate context, then move the setup logic into a before block. Something like:

```
context 'with a stored consumed_otp' do
  context 'given a precisely correct OTP'
    let(:consumed_otp) { ROTP::TOTP.new(otp_secret).at(Time.now) }

    before do
      subject.valid_otp?(consumed_otp)
    end

    it 'fails to validate' do
      expect(subject.valid_otp?(consumed_otp)).to be false
    end
  end

  context 'given a previously valid OTP within the allowed drift' do
    let(:consumed_otp) { ROTP::TOTP.new(otp_secret).at(Time.now + subject.class.otp_allowed_drift, true) }

    before do
      subject.valid_otp?(consumed_otp)
    end

    it 'fails to validate' do
      expect(subject.valid_otp?(consumed_otp).to be false
    end
  end
end
```

It make it clearer, at a glance, what part of the spec is the setup, and what part is the expectation.